server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -685,16 +727,20 @@ private static SnapshotsInRepo sortSnapshots(
                 return false;
             }
             return excludes.length == 0 || Regex.simpleMatch(excludes, policy) == false;
-        }).collect(Collectors.toUnmodifiableList());
+        };
     }
 
     private static Predicate<SnapshotInfo> filterByLongOffset(
         ToLongFunction<SnapshotInfo> extractor,
         long after,
-        String snapshotName,
-        String repoName,
+        @Nullable String snapshotName,
+        @Nullable String repoName,
         SortOrder order
     ) {
+        if (snapshotName == null) {
+            assert repoName == null : "no snapshot name given but saw repo name [" + repoName + "]";
```

Comment:
There's definitely ways of making this nicer, but since it's a temporary solution I went for the shortest+easiest-to-review version again to get this into the API (and thus complete the API side of this work) asap.